client/src/components/CreateQuestion.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div>
+    <div class="container">
+      <h2>Create question</h2>
+      <input class="input-textarea" type="text" placeholder="Question" v-model="text" />
+      <ul id="answers">
+        <li v-for="answer in answers" :key="answer.localId">
+          <input type="checkbox" :value="answer.id" v-model="correctAnswers" />
+          <input class="input-textarea" type="text" placeholder="Answer" v-model="answer.text" />
+        </li>
+      </ul>
+      <button class="signup-form-button active-theme" v-on:click.prevent="addNewAnswer">Add Answer</button>
+      <button
+        class="signup-form-button active-theme"
+        v-on:click.prevent="submitQuestion"
+      >Submit question</button>
+      <button class="signup-form-button active-theme" v-on:click.prevent="gotoQuiz">Finish</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex'
+export default {
+  name: 'CreateQuestion',
+  data () {
+    return {
+      text: '',
+      answers: [
+        {
+          id: 1,
+          text: ''
+        },
+        {
+          id: 2,
+          text: ''
+        },
+        {
+          id: 3,
+          text: ''
+        },
+        {
+          id: 4,
+          text: ''
+        }
+      ],
+      correctAnswers: []
+    }
+  },
+  props: ['quizId'],
+  methods: {
+    ...mapActions('quiz', ['addQuestion']),
+    async submitQuestion () {
+      const { text, correctAnswers, answers, quizId } = this
+      const question = { text, correctAnswers, answers }
+      const resp = await this.addQuestion({ quizId, question })
+      if (resp.status === 200) {
+        this.text = ''
+        this.answers = [
+          {
+            id: 1,
+            text: ''
+          },
+          {
+            id: 2,
+            text: ''
+          },
+          {
+            id: 3,
+            text: ''
+          },
+          {
+            id: 4,
+            text: ''
+          }
+        ]
+        this.correctAnswers = []
+      } else {
+        console.log(resp)
+      }
+    },
+    addNewAnswer () {
+      let localId = 0
+      this.answers.forEach(answer => {
+        if (answer.id > localId) localId = answer.id
+      })
```

Comment:
this could just be `this.answers[this.answers.length -1]` id

Reply:
Theoretically yes, but I had in mind an idea to be able to delete answers in the future but we dont have time for that

Reply:
if number stayed at 1 at minium there fine, besides the id could be added on request phase - you dont need it for rendering

Reply:
Reduced number to 2 in the upcoming commit